src/julia_threads.h

```diff
@@ -428,8 +442,10 @@ static inline void jl_mutex_lock_nogc(jl_mutex_t *lock)
 
 static inline void jl_mutex_lock(jl_mutex_t *lock)
 {
+    JL_SIGATOMIC_BEGIN();
```

Comment:
Sigatomic only disable the sigint safepoint and not the gc one so gc and finalizers can still be run before the lock is acquired